app/controllers/registrations_controller.rb

```diff
@@ -327,25 +329,27 @@ def stripe_webhook
     # Create a default audit that marks the event as "unhandled".
     audit_event = StripeWebhookEvent.create_from_api(event)
     audit_remote_timestamp = audit_event.created_at_remote
+    connected_account = ConnectedStripeAccount.find_by(account_id: audit_event.account_id)
+
+    if connected_account.nil?
+      logger.error "Stripe webhook reported event for account '#{audit_event.account_id}' but we are not connected to that account."
+      return head :not_found
+    end
 
     stripe_intent = event.data.object # contains a polymorphic type that depends on the event
     stored_record = StripeRecord.find_by(stripe_id: stripe_intent.id)
 
     handling_event = StripeWebhookEvent::HANDLED_EVENTS.include?(event.type)
+    incoming_event = StripeWebhookEvent::INCOMING_EVENTS.include?(event.type)
+
+    stored_record ||= StripeRecord.create_from_api(stripe_intent, {}, audit_event.account_id) if incoming_event
 
     if stored_record.nil?
-      logger.error "Stripe webhook reported event on entity #{stripe_intent.id} but we have no matching transaction."
+      logger.error "Stripe webhook reported event on entity #{stripe_intent.id} but we have no matching charge."
```

Comment:
Rename back to `transaction`, or perhaps `record`

Reply:
Fixed